extensions/lifecycle/LifecycleQueuePopulator.js

```diff
@@ -0,0 +1,127 @@
+const { usersBucket, mpuBucketPrefix } = require('arsenal').constants;
+const QueuePopulatorExtension =
+    require('../../lib/queuePopulator/QueuePopulatorExtension');
+const safeJsonParse = require('./util/safeJsonParse');
+const LIFECYCLE_BUCKETS_ZK_PATH = '/data/buckets';
+
+class LifecycleQueuePopulator extends QueuePopulatorExtension {
+
+    /**
+     * Pre-create the zookeeper path for bucket lifecycle nodes, if necessary.
+     * @param {Function} cb - The callback function.
+     * @return {undefined}
+     */
+    createZkPath(cb) {
+        const { zookeeperPath } = this.extConfig;
+        const path = `${zookeeperPath}${LIFECYCLE_BUCKETS_ZK_PATH}`;
+        return this.zkClient.getData(path, err => {
```

Comment:
as a reminder, I think this is where I put the original comment to just do the `mkdirp` directly (before github got rid of it)